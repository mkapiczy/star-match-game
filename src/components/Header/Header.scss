.App-header {
  display: flex;
  flex-direction: row;
  background-color: #3F51B5;
  flex: 0 0 auto;
  width: 100%;
  align-items: center;
  justify-content: flex-start;
  color: white;

  .app-title {
    font-size: 4vmin;
  }

  .App-logo {
    height: 4vmin;
    pointer-events: none;

    @media (prefers-reduced-motion: no-preference) {
      animation: App-logo-spin infinite 20s linear;
    }

    @keyframes App-logo-spin {
      from {
        transform: rotate(0deg);
      }
      to {
        transform: rotate(360deg);
      }
    }
  }
}