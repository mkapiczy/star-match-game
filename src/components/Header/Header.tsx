import React from 'react';
// @ts-ignore
import logo from '../../assets/logo.svg';
import './Header.scss';

const Header: React.FC = () => {
    return (
        <header className="App-header">
            <a href="/">
                <img title="logo" src={logo} className="App-logo" alt="logo"/>
            </a>
            <p className="app-title">
                Match Stars Game
            </p>
        </header>
    );
}

export default Header;