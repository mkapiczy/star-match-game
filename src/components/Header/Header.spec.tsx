import React from 'react';
import Header from "./Header";
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';

describe("Header component tests", () => {
    it("should render Header component", () => {
        const {getByText, getByTitle} = render(
            <Header/>,
        );

        const title = getByText("Match Stars Game")
        const logo = getByTitle("logo")
        expect(title).toBeInTheDocument()
        expect(logo.getAttribute("src")).toMatch("logo.svg")
    })
    it("should match snapshot", () => {
        const component = renderer.create(
            <Header/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})