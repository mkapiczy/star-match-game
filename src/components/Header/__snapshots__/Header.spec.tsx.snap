// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests should match snapshot 1`] = `
<header
  className="App-header"
>
  <a
    href="/"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
      title="logo"
    />
  </a>
  <p
    className="app-title"
  >
    Match Stars Game
  </p>
</header>
`;

exports[`Header snapshot tests should match snapshot 1`] = `
<header
  className="App-header"
>
  <a
    href="/"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
      title="logo"
    />
  </a>
  <p
    className="app-title"
  >
    Match Stars Game
  </p>
</header>
`;
